x-pack/plugins/apm/server/lib/transactions/get_transaction.ts

```diff
@@ -58,15 +54,5 @@ export async function getTransaction(
   }
 
   const resp: SearchResponse<Transaction> = await client('search', params);
-  const result = oc(resp).hits.hits[0]._source();
-
-  if (result === undefined) {
-    const notFoundError = new Error(
-      `No results found for transaction ID ${transactionId} and trace ID ${traceId}`
-    ) as HttpError;
-    notFoundError.statusCode = 404;
-    throw notFoundError;
-  }
-
-  return result;
+  return oc(resp).hits.hits[0]._source() || null;
```

Comment:
`._source(null)` is how `oc()` default values work, right?

Reply:
Yes! I wanted to do that but got:
"[ts] Argument of type 'null' is not assignable to parameter of type 'Transaction'."

As we talked about this seems like a bad check from `oc` and above was the easiest way around it.

Reply:
Okay, it is actually fixable like:
```ts
  const resp: SearchResponse<Transaction | null> = await client(
    'search',
    params
  );
  return oc(resp).hits.hits[0]._source(null);
```

But I'm not sure that's better. What do you think?

Reply:
Ah ok, I think I've run into this before too. It makes no sense that `resp` would need to account for the default value being returned from the outer function here. I am not a fan of how `ts-optchain` forces the default value to be of the same type as the deep value instead of just creating an on-the-fly union type of the `deep nested value type | default value type`. 

Anyway, I like what you had before if this is the case. :)